pass.go

```diff
@@ -0,0 +1,1054 @@
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/meh-is/pass/helpers"
+
+	"github.com/lib/pq"
+)
+
+type passConfiguration struct {
+	Addons      map[string]json.RawMessage
+	Certificate struct {
+		File string
+		Key  string
+	}
+	HSTS struct {
+		MaxAge            int
+		IncludeSubDomains bool
+		Preload           bool
+	}
+	Hostname       string
+	ListenAddress  string
+	PDOString      string
+	Secret         string
+	PasswordLength struct {
+		Min int
+		Max int
+	}
+	MaxEntryLength       int
+	SecretDays           int
+	TemplatePath         string
+	Title                string
+	TrustedProxies       []string
+	TrustedProxyIPHeader string
+}
+
+type passError struct {
+	error
+	location string
+}
+
+type passErrorInputValidation struct {
+	error
+	tooShort bool
+	tooLong  bool
+}
+
+type httpError struct {
+	error
+	code int
+}
+
+type database interface {
+	ping() (err error)
+	get(uuid string, sitePassword string) (password string, err error)
+	create(password string, sitePassword string) (uuid string, err error)
+	close()
+}
+
+type mustDatabaseConnect struct {
+	dbHandle database
+	mutex    sync.Mutex
+	connect  func() (db database, err error)
+}
+
+func (nfp *mustDatabaseConnect) doConnect() (err error) {
+	nfp.mutex.Lock()
+	defer nfp.mutex.Unlock()
+
+	if nfp.dbHandle == nil {
+		var err error
+		if nfp.dbHandle, err = nfp.connect(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+func (nfp *mustDatabaseConnect) ping() (err error) {
+	if err := nfp.doConnect(); err != nil {
+		return err
+	}
+
+	return nfp.dbHandle.ping()
+}
+func (nfp *mustDatabaseConnect) get(uuid string, sitePassword string) (password string, err error) {
+	if err := nfp.doConnect(); err != nil {
+		return "", err
+	}
+
+	return nfp.dbHandle.get(uuid, sitePassword)
+}
+func (nfp *mustDatabaseConnect) create(password string, sitePassword string) (uuid string, err error) {
+	if err = nfp.doConnect(); err != nil {
+		return "", err
+	}
+
+	return nfp.dbHandle.create(password, sitePassword)
+}
+func (nfp *mustDatabaseConnect) close() {
+	nfp.mutex.Lock()
+	defer nfp.mutex.Unlock()
+
+	if nfp.dbHandle != nil {
+		nfp.dbHandle.close()
+		nfp.dbHandle = nil
+	}
+}
+func newMustDatabaseConnect(connect func() (db database, err error)) (dbc *mustDatabaseConnect) {
+	return &mustDatabaseConnect{
+		dbHandle: nil,
+		connect:  connect,
+	}
+}
+
+type databaseConnection struct {
+	db         *sql.DB
+	createStmt *sql.Stmt
+	getStmt    *sql.Stmt
+}
+
+func (pp *databaseConnection) ping() (err error) {
+	if err := pp.db.Ping(); err != nil {
+		return err
+	}
+
+	return nil
+}
+func (pp *databaseConnection) get(uuid string, sitePassword string) (password string, err error) {
+	var result *sql.Rows
+
+	if result, err = pp.getStmt.Query(uuid, sitePassword); err != nil {
+		return "", passError{
+			error:    err,
+			location: "*databaseConnection Get pp.getStmt.Query",
+		}
+	}
+	defer result.Close()
+
+	result.Next()
+	result.Scan(&password)
+
+	return password, nil
+}
```

Comment:
Newline.

Reply:
Fixed, and database stuff moved to db.go in next push